### PR TITLE
`Development`: Add drag-and-drop exercise generation tests

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context
This PR adds tests for the validation of drag and drop exercises. In #7261 there have been alerts added to improve the UX works. The test of this PR ensure that the user keeps updated if there is input missing.

### Steps for Testing
see that all affected tests are passing 

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

**Review by laurenzfb — DISMISSED** [review]
test lgtm

**Review by aplr — DISMISSED** [review]
Looks good!

**Review by reschandreas — DISMISSED** [review]
lgtm

**Review by basak-akan — DISMISSED** [review]
test looks good 👍

File: src/test/javascript/spec/component/apollon-diagrams/apollon-diagram-detail.component.spec.ts [hunk 1]
```diff
@@ -124,6 +124,25 @@ describe('ApollonDiagramDetail Component', () => {
...
+
+        // setup
+        const div = document.createElement('div');
+        fixture.componentInstance.editorContainer = new ElementRef(div);
+        fixture.componentInstance.apollonDiagram = diagram;
+        fixture.componentInstance.initializeApollonEditor(nonInteractiveModel);
+        const errorSpy = jest.spyOn(alertService, 'error');
+
+        // test
+        await addDelay(500);
```
Comment (Hialus) [inline src/test/javascript/spec/component/apollon-diagrams/apollon-diagram-detail.component.spec.ts:138]:
```suggestion
        await fixture.componentInstance.apollonEditor?.nextRender
```
This should be the better way to do this. Documentation of `nextRender(): Promise<void>`:
> Returns a Promise that resolves when the current React render cycle is finished.

I tested this locally and it works and is faster (and on slower machines probably also more reliable) than using `addDelay`.
You might also want to consider replacing the other instances of addDelay in this (and maybe other) files.

**Review by Hialus — CHANGES_REQUESTED** [review]

File: src/test/javascript/spec/component/apollon-diagrams/apollon-diagram-detail.component.spec.ts (hunk 1 above)
Comment (milljoniaer) [reply src/test/javascript/spec/component/apollon-diagrams/apollon-diagram-detail.component.spec.ts]: Ah thank you, I should have read the Apollon Docs. But yes that's way cleaner, please have another look at the PR. 

I removed the addDelay Function and replaced it wherever it was used.

Comment (milljoniaer) [reply src/test/javascript/spec/component/apollon-diagrams/apollon-diagram-detail.component.spec.ts]: Apollon API for reference: https://apollon-library.readthedocs.io/en/stable/user/api.html#apollon-editor-api

**Review by aplr — APPROVED** [review]

**Review by reschandreas — DISMISSED** [review]

**Review by aplr — APPROVED** [review]

**Review by reschandreas — APPROVED** [review]

**Review by laurenzfb — APPROVED** [review]

**Review by basak-akan — APPROVED** [review]

**Review by Hialus — APPROVED** [review]

**Pull request merged**